cspyce/record_support.py

```diff
@@ -27,37 +45,94 @@
-    """Creates a new uninitialized record whose descriptor has the indicated name."""
-    descriptor = DESCRIPTORS.get(name)
-    if not descriptor:
-        return None
-    array = np.rec.array(None, dtype=descriptor, shape=1)
-    return array[0]
+    """Creates a new zeroed record whose descriptor has the indicated name."""
+    if name in ('SpiceEllipse', 'SpicePlane'):
+        record_type = Ellipse if name == 'SpiceEllipse' else Plane
+        return record_type()
+    else:
+        descriptor = DESCRIPTORS.get(name)
+        if not descriptor:
+            return None
+        array = np.rec.array(None, dtype=descriptor, shape=1)
+        array.fill(0)
+        return array[0]
 
 
 def as_record(name, record):
     """
-    Attempts to convert the record into a record whose descriptor has the indicated name.
+    Attempts to convert the argument into a record whose descriptor has the indicated name.
 
     If the record argument is okay, then it is returned unchanged.  Otherwise we use
     np.rec.array to try and convert it.
     """
-    descriptor = DESCRIPTORS.get(name)
-    if (descriptor is not None and isinstance(record, np.record)
-            and record.dtype == descriptor):
-        return record
-    try:
-        record = np.rec.array(record, descriptor, 1)
-        return record
-    except ValueError:
-        pass
-    # Add more attempts to convert it into a record here.
-    return None
+    if name in ('SpiceEllipse', 'SpicePlane'):
+        record_type = Ellipse if name == 'SpiceEllipse' else Plane
+        return record_type(record)
+    else:
+        descriptor = DESCRIPTORS.get(name)
+        if (descriptor is not None and isinstance(record, np.record)
+                and record.dtype == descriptor):
+            return record
+        try:
+            record = np.rec.array(record, descriptor, 1)
+            return record
+        except ValueError:
+            pass
+        # Add more attempts to convert it into a record here.
+        return None
+
+
+class _HiddenDescriptor(np.ndarray):
+    """
+    The secret subclass of np.ndarray that allows Ellipse and Plane to both be
+    numpy arrays and to have field access.
+    """
+    double_size = np.dtype(np.double).itemsize
+
+    def __new__(cls, descriptor, input_array):
+        if input_array is None:
+            assert descriptor.itemsize % cls.double_size == 0
+            input_array = np.zeros(descriptor.itemsize // cls.double_size,
+                                   dtype=np.double)
+        obj = np.asarray(input_array, dtype=np.double).view(cls)
+        obj._descriptor = descriptor
+        obj._as_record = np.rec.array(obj, dtype=obj._descriptor)[0]
+        return obj
+    
+    def as_record(self):
+        return self._as_record
+
+    def __getattr__(self, item):
+        if item in self._descriptor.names:
+            return self._as_record[item]
+        raise AttributeError
+
+
+class Ellipse(_HiddenDescriptor):
```

Comment:
Shouldn't we call these SpiceEllipse and SpicePlane to match the CSPICE library?